vars/buildPlugin.groovy

```diff
@@ -139,7 +139,25 @@ def call(Map params = [:]) {
                   mavenOptions += 'clean install'
                   try {
                     boolean usingArtifactCachingProxy = false
-                    if (artifactCachingProxyEnabled) {
+
+                    // If the build concerns a pull request, check if there is "skip-artifact-caching-proxy" label applied in case the user doesn't wan't ACP
+                    boolean prLabelsContainSkipACP = false
+                    final String skipACPLabel = 'skip-artifact-caching-proxy'
+                    if (!env.BRANCH_IS_PRIMARY) {
+                      withCredentials([
+                        usernamePassword(credentialsId: 'github-app-infra', usernameVariable: 'GITHUB_APP', passwordVariable: 'GH_TOKEN')
```

Comment:
- Need to check that this credential exists on ALL controllers (at least ci.jenkins.io :) )
- Is this credential providing the expected permissions:
  - Is installed on `jenkinsci` GitHub organization
  - Has the ability to read labels on the PRs

?